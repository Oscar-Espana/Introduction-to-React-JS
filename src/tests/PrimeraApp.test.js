import React from 'react'
import { shallow } from 'enzyme';
import PrimeraApp from '../PrimeraApp';
import '@testing-library/jest-dom'
//import { render } from '@testing-library/react'

describe('Pruebas en <PrimeraApp/>', () => {
    
    // test('debe de mostrar el mensaje "Hola, soy Oscar"', () => {
        
    //     const saludo='Hola, soy Oscar';
        
    //     const {getByText}=render(<PrimeraApp saludo={saludo}/>);
    //     expect(getByText(saludo)).toBeInTheDocument();
    // })

    test('debe de mostrar <PrimeraApp /> correctamanete', () => {
        
        const saludo='Hola, soy Oscar';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);

        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe de mostrar el subtitulo enviado por props ', () => {
        
        const saludo='Hola, soy Oscar';
        const subTitulo ='Subtitle'
        const wrapper = shallow(<PrimeraApp saludo={ saludo } subtitulo={ subTitulo } />);

        const textoParrafo=wrapper.find('p').text();
        expect(textoParrafo).toBe(subTitulo);

    })
    
    
})
