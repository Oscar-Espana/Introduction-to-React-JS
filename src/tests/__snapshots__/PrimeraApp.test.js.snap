// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> debe de mostrar <PrimeraApp /> correctamanete 1`] = `
<Fragment>
  <h1>
    Hola, soy Oscar
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
